binders/kafka-binder/spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/common/AbstractKafkaBinderHealthIndicator.java

```diff
@@ -16,27 +16,34 @@
 
 package org.springframework.cloud.stream.binder.kafka.common;
 
+import java.time.Duration;
 import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
+import java.util.Set;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
 import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.common.PartitionInfo;
 
 import org.springframework.beans.factory.DisposableBean;
 import org.springframework.boot.actuate.health.AbstractHealthIndicator;
 import org.springframework.boot.actuate.health.Health;
 import org.springframework.boot.actuate.health.Status;
 import org.springframework.boot.actuate.health.StatusAggregator;
 import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.util.Assert;
 
 /**
  * Base class that abstracts the common health indicator details for the various Kafka binder flavors.
  *
  * @author Soby Chacko
+ * @since 4.1.0
  */
 public abstract class AbstractKafkaBinderHealthIndicator extends AbstractHealthIndicator implements DisposableBean {
```

Comment:
I just noticed that in `destroy` we shutdown the executor but I don't think we are closing the consumer anywhere. I could be missing something though.